### DISCUSSION TITLE
[Login error] connection refused, Local machine runs on Windows and remote machine runs on Ubuntu 22.04(headless, not enable "Allow linux headless")

### DISCUSSION THREAD

**Discussion by Albresky** [rustdesk/rustdesk#9710] [General]
### Bug Description

On my local machine(running Windows 11 x64), I use RustDesk to connect the remote machine (running ubuntu 22.04, headless) under a local area network through `ip:port`, the RustDesk client failed to connect with a popup said "[Login error] connection refused". 

Given that when I entered the wrong password, the client would give me a message says "[Wrong password] Do you want to enter again?", which indicates the connection between my local machine and the remote works fine and the essential settings of the RustDesk in remote is configured in a corrent way.

Here is the full log from `/root/.local/share/logs/RustDesk/service/rustdesk_rCURRENT.log`:

```
[2024-10-20 18:59:30.777156 +08:00] INFO [src/core_main.rs:140] main start args: ["--service"], env: Args { inner: ["/usr/bin/rustdesk", "--service"] }
[2024-10-20 18:59:30.777448 +08:00] INFO [src/core_main.rs:262] start --service
[2024-10-20 18:59:30.892228 +08:00] INFO [src/server/uinput.rs:757] start uinput control service
[2024-10-20 18:59:30.892226 +08:00] INFO [src/server/uinput.rs:743] start uinput keyboard service
[2024-10-20 18:59:30.892254 +08:00] INFO [src/server/uinput.rs:750] start uinput mouse service
[2024-10-20 18:59:31.151488 +08:00] INFO [src/ipc.rs:318] Started ipc_service server at path: /tmp/RustDesk/ipc_service
[2024-10-20 18:59:31.151613 +08:00] INFO [src/ipc.rs:318] Started ipc_uinput_control server at path: /tmp/RustDesk/ipc_uinput_control
[2024-10-20 18:59:31.152400 +08:00] INFO [src/ipc.rs:318] Started ipc_uinput_mouse server at path: /tmp/RustDesk/ipc_uinput_mouse
[2024-10-20 18:59:31.153405 +08:00] INFO [src/ipc.rs:318] Started ipc_uinput_keyboard server at path: /tmp/RustDesk/ipc_uinput_keyboard
[2024-10-20 18:59:41.405957 +08:00] INFO [src/platform/linux.rs:310] DISPLAY: :5
[2024-10-20 18:59:41.406020 +08:00] INFO [src/platform/linux.rs:311] XAUTHORITY: /var/lib/lightdm/.Xauthority
```
---

**Possible Solutions I have tried (AFAIK)**:

- 1.  restart RustDesk service
- 2.  append `greeter-setup-script=xhost +SI:localuser:root` into /etc/lightdm/lightdm.conf
- 3.  execute `xhost +SI:localuser:root` in bash
- 4.  append `xhost +local:root > /dev/null 2>&1` into `~/.bashrc`

### How to Reproduce

Just install the nightly version of RustDesk(v1.3.1) or release version (v1.3.2).

### Expected Behavior

Establish a connection to the remote machine after entering the password.

### Operating system(s) on local side and remote side

Windows 11 -> Ubuntu 22.04

### RustDesk Version(s) on local side and remote side

1.3.1(release) -> 1.3.1(release)&1.3.2(nightly)

### Screenshots

![image](https://github.com/user-attachments/assets/31bdc1c6-7e29-4025-a6da-5ddba3ae2ff6)

**Comment by Albresky** [comment]
In previous, I use tightvnc as the remote server to connect the ubuntu and it works fine.

**Comment by Poordeveloper** [comment]
This is x11 connection refused. Something wrong or mismatch with `XAUTHORITY` and `DISPLAY`

**Reply by Albresky** [reply]
> This is x11 connection refused.

How to solve this?

**Reply by Poordeveloper** [reply]
I have no idea, because each case is different.

**Reply by Poordeveloper** [reply]
> https://github.com/rustdesk/rustdesk/issues/2573#issuecomment-1408672281

But you have tried, have you reboot or restart rustdesk service after these operations?

**Reply by Albresky** [reply]
> > [#2573 (comment)](https://github.com/rustdesk/rustdesk/issues/2573#issuecomment-1408672281)
> 
> But you have tried, have you reboot after these operations?

Yes, I have. This does not help.

**Reply by Albresky** [reply]
> > > [#2573 (comment)](https://github.com/rustdesk/rustdesk/issues/2573#issuecomment-1408672281)
> > 
> > 
> > But you have tried, have you reboot after these operations?
> 
> Yes, I have. This does not help.

A momont ago, I just found that there is an option "Allow linux headless" in the RustDesk GUI of the remote side. After enabling this option, I was able to connect to the remote on my local machine. But unfortunately, this connection only last for about 3 seconds and the connection was closed for unknown reason. And then, the local machine could not connect to the remote as we discussed before...

**Reply by Albresky** [reply]
> > > > [#2573 (comment)](https://github.com/rustdesk/rustdesk/issues/2573#issuecomment-1408672281)
> > > 
> > > 
> > > But you have tried, have you reboot after these operations?
> > 
> > 
> > Yes, I have. This does not help.
> 
> A momont ago, I just found that there is an option "Allow linux headless" in the RustDesk GUI of the remote side. After enabling this option, I was able to connect to the remote on my local machine. But unfortunately, this connection only last for about 3 seconds and the connection was closed for unknown reason. And then, the local machine could not connect to the remote as we discussed before...

Here's the process info about RustDesk running on remote machine, I don't know whether this could provide more informations for reference.

```
server-machine$ ps aux | grep rustdesk
root      369266  4.6  0.0 9186152 99328 ?       Ssl  19:41   0:15 /usr/bin/rustdesk --service
root      371352  0.0  0.0  17860  5120 ?        S    19:42   0:00 sudo -E XDG_RUNTIME_DIR=/run/user/130 -u lightdm /usr/lib/rustdesk/rustdesk --server
lightdm   371353  1.7  0.0 18651500 43008 ?      Sl   19:42   0:05 /usr/lib/rustdesk/rustdesk --server
myusername    375652  4.5  0.0 24861172 247012 ?     Sl   19:43   0:10 rustdesk
myusername    389683  0.0  0.0  12308  2048 pts/0    S+   19:47   0:00 grep rustdesk
```

**Reply by mrmeowmeow955** [reply]
i see no one else replying... so let me gonna try to help. if still needed after all those years, if connection error, there could be alot of expects...
do you have a graphical desktop envrionment like xfce4 or anything else.
is x11 | wayland server installed. 
do you have a greeter which lightdm or gdm could login with.
did you checked if youre default session-manager uses the right manager.
if its a headless server, so there is no display only ssh-access and tty terminal output.:
did you checked if you have a export -p with a active $DISPLAY output

but in youre case just from what i already see in that output:
your'e process is running :
  'sudo -E XDG_RUNTIME_DIR=/run/user/130 -u lightdm /usr/lib/rustdesk/rustdesk --server'
but i don't see no DISPLAY, XAUTHORITY and No HOME envorinments being mentiont.
the way that process should be running is like. :
sudo -u lightdm -- env XDG_RUNTIME_DIR=/run/user/110["130 in youre case, 110 is mine lightdm user"] DISPLAY=:10["in youre case output of $DISPLAY"] XAUTHORITY=/var/lib/lightdm/.Xauthority HOME=/var/lib/lightdm TERM=xterm-256color /usr/share/rustdesk/rustdesk --server

**Comment by Albresky** [comment]
Somehow I was able to establish a connection the remote, and it was closed immediately then with error code `10054`.

Here is a screenshot:
![image](https://github.com/user-attachments/assets/25a5186b-1919-4d35-9680-ae0e62f787b8)
